netkan/netkan/mirrorer.py

```diff
@@ -0,0 +1,284 @@
+        "CC-BY-ND-4.0"      : 'https://creativecommons.org/licenses/by-nd/4.0/',
+        "CC0"               : 'https://creativecommons.org/publicdomain/zero/1.0/',
+        "CDDL"              : 'https://opensource.org/licenses/CDDL-1.0',
+        "CPL"               : 'https://opensource.org/licenses/cpl1.0.php',
+        "EFL-1.0"           : 'https://opensource.org/licenses/ver1_eiffel',
+        "EFL-2.0"           : 'https://opensource.org/licenses/EFL-2.0',
+        "Expat"             : 'https://opensource.org/licenses/MIT',
+        "MIT"               : 'https://opensource.org/licenses/MIT',
+        "GPL-1.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-1.0.en.html',
+        "GPL-2.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html',
+        "GPL-3.0"           : 'http://www.gnu.org/licenses/gpl-3.0.en.html',
+        "LGPL-2.0"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.0.html',
+        "LGPL-2.1"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html',
+        "LGPL-3.0"          : 'http://www.gnu.org/licenses/lgpl-3.0.en.html',
+        "GFDL-1.1"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.1.en.html',
+        "GFDL-1.2"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.2.html',
+        "GFDL-1.3"          : 'http://www.gnu.org/licenses/fdl-1.3.en.html',
+        "LPPL-1.0"          : 'https://latex-project.org/lppl/lppl-1-0.html',
+        "LPPL-1.1"          : 'https://latex-project.org/lppl/lppl-1-1.html',
+        "LPPL-1.2"          : 'https://latex-project.org/lppl/lppl-1-2.html',
+        "LPPL-1.3c"         : 'https://latex-project.org/lppl/lppl-1-3c.html',
+        "MPL-1.1"           : 'https://www.mozilla.org/en-US/MPL/1.1/',
+        "Perl"              : 'http://dev.perl.org/licenses/',
+        "Python-2.0"        : 'https://www.python.org/download/releases/2.0/license/',
+        "QPL-1.0"           : 'https://opensource.org/licenses/QPL-1.0',
+        "W3C"               : 'https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document',
+        "Zlib"              : 'http://www.zlib.net/zlib_license.html',
+        "Zope"              : 'http://old.zope.org/Resources/License.1',
+        "Unlicense"         : 'https://unlicense.org/UNLICENSE',
+    }
+
+    def __init__(self, filename=None, contents=None):
+        super().__init__(filename, contents)
+
+    @property
+    def can_mirror(self):
+        return (
+            self.kind == 'package'
+            and self.download_content_type in CkanMirror.mime_to_extension
+            and self.redistributable
+        )
+
+    def mirrored(self, ia):
+        results = ia.search_items(self.mirror_item)
+        return True if results else False
+
+    def license_urls(self):
+        return [CkanMirror.license_url_dict[lic] for lic in self.licenses() if lic in CkanMirror.license_url_dict]
+
```

Comment:
Nit Pick: It is unusual to reference a class constant via the class name within the class.

Reply:
Well, the only way to improve common Python coding practice is one project at a time!

I want a self-documenting reminder that this is a class constant which is neither initialized per object nor a logical property of the object. The class name prefix gets me that, `self.` does not.